engine/plugins/queue/queue.js

```diff
@@ -0,0 +1,327 @@
+        });
+    };
+
+    /**
+     * Resume a queue.
+     *
+     * @param type
+     *  The type of queue (checkin or checkout).
+     */
+    resume(type) {
+        const queue = this._findQueue(type);
+        const self = this;
+
+        queue.resume().then(function() {
+            self.bus.emit('logger.info', {
+                type: 'offline',
+                message: 'Queue "' + queue.name + '" has resumed.'
+            });
+        });
+    };
+
+    /**
+     * Processing function for check-in jobs.
+     *
+     * @param job
+     *   The job.
+     * @param done
+     *   Callback when done processing.
+     */
+    checkin(job, done) {
+        const data = job.data;
+        const self = this;
+
+        // Set event callbacks that FBS should use.
+        data.busEvent = 'queue.fbs.checkout.success' + data.itemIdentifier;
+        data.errorEvent = 'queue.fbs.checkout.error' + data.itemIdentifier;
+
+        this.bus.once(data.busEvent, function(res) {
+            if (res.ok === '0') {
+                self.bus.emit('logger.err', {
+                    type: 'offline',
+                    message: require('util').inspect(res, true, 10)
+                });
+                done(new Error(res.screenMessage));
+            }
+
+            // Success the item have been checked-in.
+            done(null, res);
+        });
+
+        this.bus.once(data.errorEvent, function(err) {
+            // Log the failure.
+            self.bus.emit('logger.err', {
+                type: 'offline',
+                message: err.message
+            });
+            done(err);
+        });
+
+        // Request from the offline queue need to have the noBlock set to true to
+        // ensure that the check-in is forced in the library system.
+        data.noBlock = true;
+
+        // Set updated timestamp to get past event timeout check in FBS plugin.
+        data.timestamp = new Date().getTime();
+
+        // Send request to FBS.
+        self.bus.emit('fbs.checkin', data);
+    };
+
+    /**
+     * Processing function for checkout jobs.
+     *
+     * @param job
+     *   The job.
+     * @param done
+     *   Callback when done processing.
+     */
+    checkout(job, done) {
+        const data = job.data;
+        const self = this;
+
+        // Set event callbacks that FBS should use.
+        data.busEvent = 'queue.fbs.checkout.success' + data.itemIdentifier;
+        data.errorEvent = 'queue.fbs.checkout.error' + data.itemIdentifier;
+
+        this.bus.once(data.busEvent, function(res) {
+            if (res.ok === '0') {
+                self.bus.emit('logger.err', {
+                    type: 'offline',
+                    message: require('util').inspect(res, true, 10)
+                });
+                done(new Error(res.screenMessage));
+            }
+
+            // Success the item have been checked-out.
+            done(null, res);
+        });
+
+        this.bus.once(data.errorEvent, function(err) {
```

Comment:
Missing comment

Reply:
Don't think it is need in these calls... it is not something to do with the plugins "API" as with the specific events in the registry function.

This only an async function call... an the comment would be the same for them all..

Reply:
// Handle error event.